### ISSUE TITLE
CSS: handing the a11y of anchor positioning

### ISSUE THREAD

**Issue by scottaohara** [w3c/html-aam#545]
briefly spoke about this with @aleventhal today (cc @mfreed7)

from what i understand there have been talks about creating an aria-details relationship, similar to the one defined for popovers, if authors were to use CSS anchor positioning to associate elements with one another.

this may be useful / help provide a baseline of information for instances where an author has decided to 'tether' one element to another in the UI, but not provide any meaningful semantics to programmatically communicate this relationship.

if this were to be done, would expect this to have similar logic to popovers, where the aria-details relationship is not exposed if the elements are direct accessibility siblings to each other.  Additionally, it might make sense to not expose this relationship depending on the roles of the elements that are associated.  e.g., if someone were to use a role=tooltip and/or had an aria-describedby association already made, then the aria-details association would likely just be extra noise - so maybe don't do that then.

also, might be worth reiterating that if someone were to use an `aria-details=""` on one of the elements, that the details relationship should be considered null - as that should be considered a signal from the author that they do not want there to be a details relationship, for whatever reason.

**Comment by aleventhal** [comment]
+1 on copying the logic from popover siblings, and to not doing it for
aria-details=""

**Comment by smhigley** [comment]
That makes sense to me. I think it'd be good to expand the list of cases where the relationship would not be added -- does this list make sense to you two?:
- `aria-labelledby`
- `aria-describedby`
- `aria-details` (would this be excluded just with an empty string, or for any existing details relationship?)
- `aria-owns`

**Comment by aleventhal** [comment]
Hi Sarah, interesting, I hadn't thought of these.
After mulling it over, I think that probably only aria-details needs to
prevent aria-details from being auto-added. Well that, and if the detail
thing would have been the next sibling anyway (same rule as for popover).
For example, just because something has a label or description doesn't mean
that another area of the page being pointed to is irrelevant.
As far as aria-owns, these should be treated like any other children,
shouldn't they? Maybe we can imagine use cases. Since I don't understand
this one very well, I'm tempted to leave it out as well.
In the end I think a clear rule where we aren't guessing a lot might be
better, even if we get it wrong sometimes.
So my vote so far is to go with the same rule as for popovers. WDYT?

**Comment by smhigley** [comment]
@aleventhal sorry, I think I should've been more specific 😅. I didn't mean if the anchor had any `aria-labelledby`/`aria-describedby` at all, but specifically if it already had that relationship with the anchored element. In that case, I think it would be good to not make the anchored element into both the label and the details of the anchor.

My thought behind `aria-owns` is similar to excluding it for siblings -- if the anchored element is already a child or owned element of the anchor, then it probably doesn't also need a details relationship. I suppose that could get a little iffy for containers with a lot of content, though (also not sure how common that is).

**Comment by aleventhal** [comment]
Oh ok, all of this makes sense.

**Comment by spectranaut** [comment]
Discussed in today's meeting: https://www.w3.org/2024/05/30-aria-minutes.html#t06

**Comment by scottaohara** [comment]
@pkra likely not - as the details relationship isn't present when popovers are not rendered.

**Comment by jpzwarte** [comment]
I've created https://issues.chromium.org/issues/423841930 which is related to this. In this codepen (https://codepen.io/jpzwarte/pen/MYwEaVm), `aria-details` of the `<button popovertarget>` points to a purely visual, non semantic `<div class="tether">` element.

**Comment by pkra** [comment]
Now that we have two implementations, could we revisit this?

**Comment by keithamus** [comment]
Right. There wasn't a strong conclusion from the meeting at TPAC. I still believe it's not totally the right choice to _always_ expose details for _any_ anchor. Just over lunch I was speaking to a developer who said they were customizing the anchor of a button inside a `<td>` to instead anchor on the parent `<tr>` so that the floating UI appears over the whole row. Adding that association doesn't seem useful, to me (though happy to be told how it actually would be). 

@smhigley mentioned using the anchor relationship as a fallback; so that if an anchor element does not have a pre-existing details, labelled, described, or owns relationship then it could fall back to the anchor. I would like to see use cases for this; does it make a page that was otherwise less accessible more accessible?
